src/sql/pool.rs

```diff
@@ -35,6 +79,43 @@ impl InnerPool {
         connections.push(connection);
         drop(connections);
     }
+
+    /// Retrieves a connection from the pool.
+    ///
+    /// Sets `query_only` pragma to the provided value
+    /// to prevent accidentaly misuse of connection
+    /// for writing when reading is intended.
+    /// Only pass `query_only=false` if you want
+    /// to use the connection for writing.
+    pub async fn get(self: Arc<Self>, query_only: bool) -> Result<PooledConnection> {
+        let permit = self.semaphore.clone().acquire_owned().await?;
+        let conn = {
+            let mut connections = self.connections.lock();
+            connections
+                .pop()
+                .context("got a permit when there are no connections in the pool")?
+        };
+        if query_only {
+            conn.pragma_update(None, "query_only", "1")?;
+            let conn = PooledConnection {
+                pool: Arc::downgrade(&self),
+                conn: Some(conn),
+                _permit: permit,
+                _write_mutex_guard: None,
+            };
+            Ok(conn)
+        } else {
+            let write_mutex_guard = Arc::clone(&self.write_mutex).lock_owned().await;
```

Comment:
So we can take a connection and then sleep here until another writer finishes. I suggest not to reorder locking and acquiring a connection